### PR TITLE
CLOUDP-81627: Improve Atlas cluster create dynamic version for limited internet envs

### PR DESCRIPTION
**Pull request by gssbzn**
## Proposed changes

_Jira ticket:_ CLOUDP-81627

## Checklist

- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added any necessary documentation (if appropriate)
- [x] I have run `make fmt` and formatted my code

## Further comments

Default timeouts were too long for users using ops manager with no access to Atlas and just doing `mongocli --version` could take a long time

I'm handling this in two ways, not longer use of default timeouts and also lowering some of the ones we had. and we'll no longer include the Atlas branch if we don't need to, saving the call completely

### CODE REVIEW

File: cmd/mongocli/mongocli.go [hunk 1]
```diff
@@ -106,5 +118,8 @@ func initConfig() {
 }
 
 func main() {
+	cobra.EnableCommandSorting = false
+	cobra.OnInitialize(initConfig)
```
Comment (andreaangiolillo) [inline cmd/mongocli/mongocli.go:122]: [q] I am not sure why we need these two lines

File: cmd/mongocli/mongocli.go [hunk 2]
```diff
@@ -65,30 +67,40 @@ no additional shell configuration is necessary, see https://docs.brew.sh/Shell-C
...
+	if len(argsWithoutProg) != 0 && argsWithoutProg[0] == "--version" {
+		return
+	}
+	rootCmd.AddCommand(cliconfig.Builder())
 
-func init() { //nolint:gochecknoinits // This is the cobra way
+	// We realistically only care about not adding Atlas to the chain of commands
+	// but given we can reuse the pattern for the rest we can save some initialization time here
+	if len(argsWithoutProg) == 0 || argsWithoutProg[0] == atlas.Use {
+		rootCmd.AddCommand(atlas.Builder())
```
Comment (andreaangiolillo) [inline cmd/mongocli/mongocli.go:89]: [q] how do you know that the first argument is the service? This question may be related to the fact that I don't know what `[1:]` returns.

**Review by andreaangiolillo — APPROVED** [review]
LGTM. Just a couple of questions

File: cmd/mongocli/mongocli.go (hunk 1 above)
Comment (gssbzn) [reply cmd/mongocli/mongocli.go:122]: those lines used to be in L87 and L91 I just got rid of the init function
cobra by default sorts command alphabetically, we prefer to use our own order, so `cobra.EnableCommandSorting`

Also we call `initConfig` to initialize viper

File: cmd/mongocli/mongocli.go (hunk 2 above)
Comment (gssbzn) [reply cmd/mongocli/mongocli.go:89]: if you call the cli like `mongocli atlas clusters` then `os.Args` is `["mongocli", "atlas", "clusters"]`, `os.Args[1:]` would be `["atlas", "clusters"]`

Comment (andreaangiolillo) [reply cmd/mongocli/mongocli.go:89]: thanks!!!

**Pull request merged**